src/test/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGANaivelySparkUnitTest.java

```diff
@@ -0,0 +1,278 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import org.apache.commons.lang3.StringUtils;
+import htsjdk.samtools.fastq.FastqReader;
+import htsjdk.samtools.fastq.FastqRecord;
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.engine.ReadsDataSource;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.BaseUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.testng.Assert;
+import org.testng.SkipException;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import scala.Tuple2;
+
+import java.io.*;
+import java.io.FileNotFoundException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+
+// TODO: a caveat that despite exact same inputs and outputs for all previous steps before sga overlap,
+//   the final assembled contigs, run on Broad gsa machines and on travis, will give "slightly" different results
+//   for--in this unit test case--the contigs of length 1644 and 1720; so we're testing on a slightly less stringent level:
+//   that is, assembled contigs are allowed to be a certain edit distance away from the expected contigs.
+// TODO: test if error messages are handled correctly
+public class RunSGANaivelySparkUnitTest extends CommandLineProgramTest {
+
+    private static final File TEST_DATA_DIR = new File(getTestDataDir(), "spark/sv/RunSGANaivelySpark/");
+    private static final Path sgaPath;
+    static {
+        if (System.getProperty("user.name").equalsIgnoreCase("shuang")) {
+            sgaPath = Paths.get("/usr/local/bin/sga");
+        } else {
+            sgaPath = null; // for testing on machines that doesn't have SGA installed, skip test
+        }
+    }
+
+    private static final int editDistanceTolerance = 2;
+
+    private static final int threads = 1;
+
+    private static final SGAModule indexer = new SGAModule("index");
+
+    private static Tuple2<Long, Iterable<GATKRead>> breakpointIDToReads;
+    private static Tuple2<Long, File> tempFASTQFile;
+    private static final File workingDir;
+    static {
```

Comment:
You should be able to do this stuff in a setup() test method I think, rather than a static initializer.